analyzers/src/SonarAnalyzer.CFG/Extensions/ControlFlowRegionExtensions.cs

```diff
@@ -42,7 +42,9 @@ public static ControlFlowRegion EnclosingRegionOrSelf(this ControlFlowRegion reg
         {
             while (region is not null && region.Kind != kind)
             {
-                region = region.EnclosingRegion;
+                region = region.Kind == ControlFlowRegionKind.Root
+                    ? null  // Do not traverse from inner lambda CFG to the outer method CFG
+                    : region.EnclosingRegion;
```

Comment:
Personal opinion: I think this is easier to follow
```cs
if (region.Kind == ControlFlowRegionKind.Root)
{
  return null; // Do not traverse from inner lambda CFG to the outer method CFG
}
region = region.EnclosingRegion;
```

Reply:
Have you checked other potentially dangerous ControlFlowRegionKinds?

Reply:
Others are not dangerous, only the root should be the stopping point. I didn't know that we can breach out of the root before.